### ISSUE TITLE
Doesn't show up in KTp

### ISSUE THREAD

**Issue by LuccoJ** [abderrahim/telepathy-poison#1]
Kind of not-a-real-issue since this isn't expected to work, I guess... but it's pretty cool, and I hope you keep developing it!
I got as far as connecting it to my existing Tox profile with "mc-tool", and having it show up as online. But then, when I try adding a contact from my KDE Telepathy contact list, the Tox account doesn't show up, while the in-panel contact list crashes completely when I try opening it with Poison running.

Want any debugging output?

**Comment by abderrahim** [comment]
I have only tested it with gnome/empathy, and had to work around some crashes, so I expect it to crash with other implementations.

If I understand correctly, when you try to add a contact, you're not offered the option to add a contact for your tox account. Do existing contacts show there?

I'm not sure about debugging output. From my experience, logs from [bustle](http://www.freedesktop.org/wiki/Software/Bustle/) can be useful to see whether it's calling a d-bus method I didn't implement; otherwise, especially for the crash it is best to see the source code to find what it expects that poison isn't doing.

Anyway, you can try to get a d-bus log with bustle and debugging output by running
`env G_MESSAGES_DEBUG=all telepathy-poison > telepathy-poison.log`
but I'm not overconfident they will be helpful.

**Comment by LuccoJ** [comment]
Yesterday, even without G_MESSAGES_DEBUG=all, I was already getting some warnings from the telepathy-poison output.
Today perhaps I did something different, but the contact list is just going black and stops working after I get my Tox account running in telepathy-poison.

However, is it possible perhaps that there is a problem with the "available" (2) status not being given as "available" by telepathy-poison but as an empty string? This is my "mc-tool show poison/tox/account0" output:

Account: poison/tox/account0
Display Name: LjL
Normalized: FDC973C0BBB4A4B264E601FE145C5396DBD62CD2C41F042D2E4D.....
Enabled: enabled
Icon: im-tox
Connects: only when requested
Nickname: LjL
Service: tox

Presences:
Automatic: available (2) ""
Current:  (2) ""
Requested: available (2) ""
Changing: yes

(string) profile = LjL

"Requested" and "Current" are both 2, but "Current" seems to have no string associated to it... and this seems to match the fact that in my contact list program, I now see an empty box instead of the text "Available".

**Comment by LuccoJ** [comment]
Ah, now I asked for "mc-tool request poison/tox/account0 online" instead of "mc-tool request poison/tox/account0 available", and the behavior is different - more like yesterday's behavior, and I get this output from telepathy-poison whenever I try to access my contact list program:

*\* (process:13777): CRITICAL **: connection_real_request_handles: assertion '_tmp1_ == ((guint) TELEPATHY_HANDLE_TYPE_CONTACT)' failed
org.freedesktop.Telepathy.Channel.ChannelType -> 'org.freedesktop.Telepathy.Channel.Type.ContactList'
org.freedesktop.Telepathy.Channel.TargetHandleType -> uint32 3
org.freedesktop.Telepathy.Channel.TargetHandle -> uint32 0

In both cases, when I try to add a contact from my contact list program, I only see my existing XMPP accounts, and no Tox.

**Comment by LuccoJ** [comment]
I can confirm, by the way, that this works (connects, at least, I didn't chat with anyone) on Empathy, I just installed GNOME and tried. I can seemingly add contacts - not tried to add one, but the account shows up and it prompts me to add contacts, unlike on KDE.
Good job!

I don't know why it's different with KDE. That error message I posted above doesn't happen in GNOME.

One tiny issue when compiling: the "data" directory where the .service file gets put doesn't exist, and unless I make it manually, "make" fails.

**Comment by abderrahim** [comment]
Ok, so kde-telepathy is trying to get the contact list the old way. It's marked as deprecated in the telepathy specification, so I didn't bother implementing it, but it may be necessary after all.

BTW, what should be working now is adding an account (from the gui, no need for mc-tool), sending and receiving messages and sending and receiving contact requests. I've added a d-bus activation support (the .service file), so it doesn't need to be run manually, but it doesn't seem to work for now.

**Comment by LuccoJ** [comment]
I didn't try adding an account from the GUI since it definitely wouldn't work in KDE, so I didn't even think about trying it in GNOME. The DBus .service seemed to work for me in GNOME... I mean, yesterday I logged out and back in, started Empathy without starting telepathy-poison manually, and I already had my Tox contacts showing there. I assumed that meant telepathy-poison had been automatically started.

I cannot test on GNOME anymore for now because I only have GNOME on my desktop and I'll be away from home for a few days. I can test things under KDE, though, or I suppose I can test other Telepathy clients without too many dependencies (I have capped data here).
